pkg/policy/policy.go

```diff
@@ -204,182 +137,57 @@ func (p *Policies) Applicable(resource client.Object) (bool, string, error) {
-		metadata, err := rego.New(
-			rego.Compiler(compiler),
-			rego.Query(metadataQuery),
-		).Eval(ctx)
-		if err != nil {
-			return nil, fmt.Errorf("failed evaluating Rego metadata rule: %s: %w", metadataQuery, err)
-		}
-
-		metadataResult, hasMetadataResult := hasBinding(metadata, varMetadata)
-
-		if !hasMetadataResult {
-			return nil, fmt.Errorf("failed parsing policy metadata: %s", policyName)
-		}
-
-		md, err := NewMetadata(metadataResult)
-		if err != nil {
-			return nil, fmt.Errorf("failed parsing policy metadata: %s: %w", policyName, err)
-		}
-
-		denyQuery := fmt.Sprintf("%s.deny[res]", parsedPolicy.Package.Path.String())
-		deny, err := rego.New(
-			rego.Compiler(compiler),
-			rego.Query(denyQuery),
-			rego.Input(resource),
-		).Eval(ctx)
-
-		if err != nil {
-			return nil, fmt.Errorf("failed evaluating Rego deny rule: %s: %w", denyQuery, err)
-		}
-
-		denyValues, hasDenyValues := hasBindings(deny, varResult)
-		if hasDenyValues {
-			denyResults, err := valuesToResults(md, denyValues)
-			if err != nil {
-				return nil, fmt.Errorf("failed parsing deny rule result: %s: %w", denyQuery, err)
-			}
-			results = append(results, denyResults...)
-			continue
-		}
-
-		warnQuery := fmt.Sprintf("%s.warn[res]", parsedPolicy.Package.Path.String())
-		warn, err := rego.New(
-			rego.Compiler(compiler),
-			rego.Query(warnQuery),
-			rego.Input(resource),
-		).Eval(ctx)
-
-		if err != nil {
-			return nil, fmt.Errorf("failed evaluating Rego warn rule: %s: %w", warnQuery, err)
-		}
-
-		warnValues, hasWarnValues := hasBindings(warn, varResult)
-		if hasWarnValues {
-			warnResults, err := valuesToResults(md, warnValues)
-			if err != nil {
-				return nil, fmt.Errorf("failed parsing warn rule result: %s: %w", warnQuery, err)
-			}
-			results = append(results, warnResults...)
-			continue
-		}
-
-		results = append(results, Result{
-			Metadata: md,
-			Success:  true,
-		})
-	}
-
-	return results, nil
-}
-
-func hasBinding(rs rego.ResultSet, key string) (map[string]interface{}, bool) {
-	if rs == nil || len(rs) == 0 {
-		return nil, false
-	}
-	binding, ok := rs[0].Bindings[key]
-	return binding.(map[string]interface{}), ok
-}
-
-func hasBindings(rs rego.ResultSet, key string) ([]map[string]interface{}, bool) {
-	if rs == nil || len(rs) == 0 {
-		return nil, false
-	}
-	var values []map[string]interface{}
-
-	for _, r := range rs {
-		binding, ok := r.Bindings[key]
-		if !ok {
-			continue
-		}
-		value, ok := binding.(map[string]interface{})
-		if !ok {
-			continue
-		}
-		values = append(values, value)
+	memfs := memoryfs.New()
+	// add policies files
+	err = createPolicyInputFS(memfs, policiesFolder, policies, "rego")
+	if err != nil {
+		return nil, err
 	}
-	return values, len(rs) == len(values)
-}
-
-func requiredStringValue(values map[string]interface{}, key string) (string, error) {
-	value, ok := values[key]
-	if !ok {
-		return "", fmt.Errorf("required key not found: %s", key)
+	inputResource, err := json.Marshal(resource)
```

Comment:
Can we run json marshal before declaring the const on line 147? if we can't marshal for some reason, there is no need get the policies and to create a memoryfs.

Reply:
I prefer to declare `inputResource` close to its usage L167